src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -611,6 +614,8 @@ protected override void Dispose(bool disposing)
                         base.Dispose(disposing);
                     }
                 }  // finally
+
+                s_byteBufferPool.Free(_buffer);
```

Comment:
Pool limits the number of retained objects, but you also need to ensure a limit for the size of stored objects. Otherwise the cache is unbounded - also known as memory leak.